pageserver/src/waldecoder.rs

```diff
@@ -731,7 +814,7 @@ pub fn decode_wal_record(record: Bytes) -> DecodedWALRecord {
     // 5. Handle a few special record types that modify blocks without registering
     // them with the standard mechanism.
     if xlogrec.xl_rmid == pg_constants::RM_HEAP_ID {
-        let info = xlogrec.xl_info & pg_constants::XLOG_XACT_OPMASK;
+        let info = xlogrec.xl_info & pg_constants::XLR_RMGR_INFO_MASK;
```

Comment:
Is this an existing bug? Better fix it as a separate PR  if it is

Reply:
I hoped that it was fixed long ago.
Will create PR for it.